packages/babel-core/src/config/full.js

```diff
@@ -278,17 +278,32 @@ const instantiatePlugin = makeWeakCache(
   },
 );
 
+const validatePreset = (
+  preset: PresetInstance,
+  context: ConfigContext,
+  descriptor: UnloadedDescriptor,
+): void => {
+  if (!context.filename) {
+    const { options } = preset;
+    if (options.test || options.include || options.exclude) {
```

Comment:
We should also check if the `overrides` option contains a block which has one of these three options (like in the TS preset).

Reply:
Good catch. Fixed in 056e197.